src/utils.py

```diff
@@ -202,3 +202,36 @@ def end_time_measurement(start: float) -> float:
     """
     end = time.perf_counter()
     return round((end - start) * 1000, 3)
+
+
+def quark_stop_watch(position: int = None):
+    """
+    Usage as decorator to measure time, eg:
+    ```
+    @quark_stop_watch()
+    def run(input_data,...):
+        return processed_data
+    ```
+    results in valid:    
+    ```
+    processed_data, time_to_process = run(input,...)
+    ```
+    If the return value of the decorated function is of type tuple,
+    the optional parameter `position` can be used to specify the position at which the
+    measured time is to be inserted in the returned tuple.
+
+    :param position: the position at which to insert the time
+    :type position: int
```

Comment:
Please add a parameter and type description for the output as well

Reply:
done. But note that the user of the decorator is usually not interested in the return value of the decorator but in the return value of the decorated function.